### ISSUE TITLE
[LC/AON Timer/Pwrmgr] AON Timer stops counting after low power entry

### ISSUE THREAD

**Issue by sriyerg** [lowRISC/opentitan#6505]
Here's the waveform snippet:
![image](https://user-images.githubusercontent.com/46467186/117948006-9733fc00-b2c5-11eb-95be-064f10e14676.png)

Found when debugging the `dif_pwrmgr_smoketest.c` running in DV environment. 

Related bugs: #6503, #6502. 
[Too late for me to dig further].

**Comment by sriyerg** [comment]
[waves.fsdb.zip](https://github.com/lowRISC/opentitan/files/6464926/waves.fsdb.zip)

Attaching waves - should be able to convert to VPD to debug with GTKWave.

**Comment by tomroberts-lowrisc** [comment]
Couldn't figure out a way of converting the fsdb, but I managed to reproduce the failure.

The timer stops because the lifecycle state has been scrapped. Spent a bit of time poking around but couldn't figure out why it's happening. Maybe @msfschaffner could have a look?

**Comment by sriyerg** [comment]
When pwrmgr initiates the low power entry, it puts the LC in reset, which causes `lc_cpu_en_o` to be disabled. The AON timer looks at that signal to know we are in scrap state and hence we no longer need to count.

**Comment by msfschaffner** [comment]
This seems to be an artifact of https://github.com/lowRISC/opentitan/pull/5480, where we switched from `lc_escalate_en` to `lc_cpu_en` in order to enable/disable the AON timer. The change was motivated by the fact that `lc_escalate_en` was not asserted in certain LC SCRAP/INVALID states.

However, in the mean time, we have also changed the `lc_escalate_en` behavior such that it is always asserted in LC SCRAP/INVALID states.
Hence, in order to get around the issue we are currently experiencing when entering deep sleep, a viable solution would be to revertback to using `lc_escalate_en` to disable the AON timer: https://github.com/lowRISC/opentitan/pull/6511
